Product Filters: only query selected attributes for attribute query clauses

dinhtungdu opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

<!-- For bug fixes: If known, please provide links to help with traceability and escape analysis. -->
<!-- Please include a link to the issue of the bug being fixed, if one doesn't exist please create it. -->
<!-- If the PR that introduced the bug is known, please also add its link below. -->

Closes [WOOPLUG-4936](https://linear.app/a8c/issue/WOOPLUG-4936/improve-performance-of-product-filters-by-querying-only-selected).

This PR fixes the logic querying attribute terms for attribute query clause. Instead of query all terms, we query only selected terms to improve the performance and reduce the unnecessary logic for filtering out unselected term later.

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

There should be no regression related to product filtering around attribute. Smoke test the Product Filters block with multiple filter combination to ensure the Attribute filter is still working as expected.

<!-- End testing instructions -->

dinhtungdu committed:
perf: optimize attribute term retrieval in QueryClauses

- Replace buggy get_terms() call with proper taxonomy and slug parameters
- Use single query to fetch all attribute terms instead of inefficient filtering
- Group terms by taxonomy for better performance
- Remove unnecessary get_current_attribute_terms() helper method
- Improves performance when multiple attribute filters are applied

dinhtungdu committed:
chore: changelog

gigitux reviewed (approved):
LGTM! 🚢 

I noticed that there are two changelogs. Can you remove one before merging the PR? Thanks!

dinhtungdu committed:
fix: remove duplicated changelog

dinhtungdu committed:
fix: defense!

dinhtungdu committed:
fix: improve safety of array operations in attribute term query

dinhtungdu committed:
refactor: use empty() instead of isset() for terms validation

Using empty() is more appropriate as it skips empty arrays which have no
filtering value, making the query more efficient and semantically correct.
